### DISCUSSION TITLE
Show approximate time duration in time picker

### DISCUSSION THREAD

**Discussion by justinsmalley** [grafana/grafana#66066] [Dashboards, Panels & Visualizations]
When I am exploring a dashboard, zooming in on a time range, zooming out, shifting time forward and backward, I sometimes lose track of the approximate time frame I am working with. I.e. am I looking at an hour of data? a day? a month? It figure this out I have to look at the time values in the top right and perform a mental calculation of how far apart the dates are.

What would be helpful is to next to the dates in the top right give an approximation of the duration. 

Here is a way it could possibly look, although I’m sure grafana designers can come up with something that looks better.
![image](https://user-images.githubusercontent.com/99200643/230198909-f210be1f-2806-4d34-87c1-f12bfc5f4500.png)

To reduce space and make it easier to ingest at a glance, it could be limited to one time scale (e.g. seconds, hours, months, etc.) and one decimal point. For example, if the time difference is 4830s = “1h, 20m, 30s”, it could be approximated down to “~1.3h”. The decimal could even be dropped as the number gets higher. For example, 1.5 days is more valuable than 1 day, but 8.6 days is not adding any additional information than simply saying 9 days. The keys that I am trying to communicate is we want the fewest characters (ideally between 2 and 4) to give a rough scale of what I am looking at presently on the dashboard.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82784. Feel free to continue the discussion around this there. Thank you!
